Reland "sliverGridDelegate mainAxisExtent add assert (#148470)"

hello-coder-xu opened the pull request:
Reland: https://github.com/flutter/flutter/pull/148470
Fixes https://github.com/flutter/flutter/issues/138871

hello-coder-xu committed:
merged master code

hello-coder-xu committed:
merged master code

hello-coder-xu committed:
format code

hello-coder-xu committed:
format code

Piinks commented:
This was reverted after https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20web_canvaskit_tests_7_last/14617/overview failed in postsubmit. 

Previously it passed in presubmit, and it passing again in presubmit now. If it fails again and this is reverted again, we should file an infra ticket to investigate.

Piinks reviewed (approved):
LGTM

justinmc commented on packages/flutter/test/widgets/grid_view_test.dart:
```diff
@@ -898,4 +898,60 @@ void main() {
     expect(controller.position.maxScrollExtent, 472.0);
     expect(controller.position.pixels, 472.0);
   });
+  testWidgets('SliverGridDelegate mainAxisExtent add assert', (WidgetTester tester) async {
```
Nit: Put a newline above this line.

hello-coder-xu replied:
updated

justinmc reviewed (approved):
LGTM :+1:

hello-coder-xu committed:
add newline

hello-coder-xu committed:
add newline

hello-coder-xu committed:
remove space

hello-coder-xu committed:
remove space

Piinks commented on packages/flutter/test/widgets/grid_view_test.dart:
```diff
@@ -898,4 +898,61 @@ void main() {
     expect(controller.position.pixels, 472.0);
   });
+
+  testWidgets('SliverGridDelegate mainAxisExtent add assert', (WidgetTester tester) async {
```
This tests is currently failing on web, this is why it was reverted previously, Can you take a look?

hello-coder-xu committed:
Merge branch 'master' into feature/reland-sliverGridDelegate-mainAxisExtent-add-assert

* master:
  Revert "Reland 2: [CupertinoActionSheet] Match colors to native" (#150142)
  Reland 2: [CupertinoActionSheet] Match colors to native (#150129)

hello-coder-xu committed:
update test impl

hello-coder-xu committed:
update test impl

Piinks commented:
Hmm, it looks like some tests are still failing here.

hello-coder-xu committed:
Merge branch 'feature/reland-sliverGridDelegate-mainAxisExtent-add-assert' of https://github.com/hello-coder-xu/flutter into feature/reland-sliverGridDelegate-mainAxisExtent-add-assert

* 'feature/reland-sliverGridDelegate-mainAxisExtent-add-assert' of https://github.com/hello-coder-xu/flutter:
  update test impl
  remove space
  add newline
  format code
  merged master code
  Update testowners (#150141)

hello-coder-xu committed:
udpate test impl

hello-coder-xu commented:
> Hmm, it looks like some tests are still failing here.


I pay attention to your replies every day. I refer to other assertion test methods, but they still fail. My local test can pass normally (packages/flutter).
The code of reland has not changed much from before. It appeared after merging the latest master code. I have been busy with work recently, so I will take time to deal with it.
Thank you for your guidance. I will reply to you after confirming that it has been handled.

hello-coder-xu commented:
@Piinks 
I changed the error content matcher. The previously pushed code could not be passed. It should be Linux that escaped the single quotes in "'mainAxisExtent == null || mainAxisExtent >= 0' ", and finally became "\'mainAxisExtent == null || mainAxisExtent >= 0\' "
I removed the single quotes
